lib/db.js

```diff
@@ -315,29 +328,36 @@ const collectionKeys = [
  * @param {boolean} [options.serializeFunctions=false] Serialize functions on any object.
  * @param {boolean} [options.strict] Returns an error if the collection does not exist
  * @param {object} [options.readConcern] Specify a read concern for the collection. (only MongoDB 3.2 or higher supported)
- * @param {object} [options.readConcern.level='local'] Specify a read concern level for the collection operations, one of [local|majority]. (only MongoDB 3.2 or higher supported)
  * @param {Db~collectionResultCallback} [callback] The collection result callback
  * @return {Collection} return the new Collection instance if not in strict mode
  */
+const collectionSchema = {
+  w: { type: ['number', 'string'] },
+  wtimeout: { type: 'number' },
+  j: { type: 'boolean' },
+  raw: { type: 'boolean' },
+  pkFactory: { type: 'object' },
+  readPreference: { type: [ReadPreference, 'object', 'string'] },
```

Comment:
why the extra `object` here?

Reply:
We need this `object` to handle `readPreference`s that are specified as objects, like in `Should correctly apply readPreferences specified as objects` in `readpreference_tests`, where the `readPreference` is `{ mode: 'secondaryPreferred', tags: [] }`.

Reply:
bummer, did you end up making a ticket to track deprecating this form?

Reply:
Yep, I made NODE-1752.